### PR TITLE
📖 Enable RTD again

### LINKED ISSUE #1133: ReadTheDocs hosting support
When a user creates a page with RTD, they are currently encountering some issues. Below is a list of tasks that should be supported:

- [ ] The landing page of the preview is broken, due to some kind of javascript error
- [ ] Bare links need to redirect to `.html` files
- [ ] Overwrite `robots.txt`
- [ ] Handle hydration issues
- [ ] Add `flyout` menu support

### PR DESCRIPTION
**Pull request by LecrisUT**
Let's keep this in draft so we can explore how to fix the RTD issues. Quick comment @rowanc1 can you keep the RTD project alive so that it can run on PRs. It is fine if it fails on main/other PRs due to a lack of `.readthedocs.yaml` file, it's just that it would be good to have it here so that we can explore all of the issues and potential solutions across the `myst-theme` repo (i.e. open a similar branch there and point `packages.json` to the repo branch where we can explore potential fixes)

TODO:
- [x] Write todo
- [ ] The landing page of the preview is broken, due to some kind of javascript error
  @choldgraf Can you comment with a screenshot of this? 
- [ ] Bare links need to redirect to `.html` files: `myst-to-react` issue
  - Temporary solution would be to have the hosting provider do the redirect, but that is not ideal because local navigation of the static site through `index.html` is still broken
- [ ] Overwrite `robots.txt`
  - Maybe this is to be done on the RTD settings?
- [ ] Handle hydration issues: `myst-to-react` issue
- [ ] Add `flyout` menu support

Closes #1133

### CODE REVIEW

File: .readthedocs.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+build:
+  os: ubuntu-lts-latest
+  tools:
+    nodejs: "20"
+  commands:
+    - npm install
+    - npm run build
+    - npm run --workspace packages/mystmd link
+    - myst --version
+    - cd docs && myst build --html
```
Comment (LecrisUT) [inline .readthedocs.yaml:11]:
```suggestion
    - cd docs && myst build --html --execute
```

Last time it was suggested to use `--execute`, but my understanding is that only affects the running of notebooks. For these issues it should not be relevant right? Is it also used in the mystmd pages?

**Comment by rowanc1** [comment]
Thanks @LecrisUT for reopening this. @choldgraf I think you turned off the preview on RTD, are you in a place to re-enable?

**Comment by choldgraf** [comment]
I can do that but just want to make sure that we are ready to do it. If I enable it then GitHub will mark every PR as failing until the rtd job is re enabled. Do we want that now or do we want to wait until we are actively fixing this before doing so?

**Comment by LecrisUT** [comment]
Oh fair point. I can make a small page to redirect here that would be run on main and all other branches

**Comment by humitos** [comment]
You can activate this branch in RTD without enabling the pull request build if you want. That way, this branch will be built on every commit on RTD... Getting an outcome similar to the PR builds.

**Comment by humitos** [comment]
> Overwrite robots.txt

You can do this by generating this file as you want and saving it at the root of the HTML output directory. More at https://docs.readthedocs.io/en/stable/reference/robots.html

**Comment by LecrisUT** [comment]
It would not have the nice PR link to show it, and it would have to be on a new branch on the main repo, but that would work

**Comment by agoose77** [comment]
Given that we need to do specific work #984 #997 for RTD support, I'm closing this for now. Do reply here if you think there's good reason to keep this particular PR open in the near term!

**Pull request closed without merging**